File: frontend/types/recoveryPlan.ts

export type RecoveryPlan = {
    id: number;
    name: string;
    workoutDays: WorkoutDay[];
    isCreatedByDoctor: boolean;
};

export type WorkoutDay = {
    dayNumber: number;
    exercises: Exercise[]; 
};

export type Exercise = {
    id: number;
    name: string;
    description: string;
    sets: number;
    reps: number;
    duration: string | null; 
};

export type RecoveryPlanWithAppUserInfo = {
    id: number;
    name: string;
    workoutDays: WorkoutDay[];
    appUserId: string;
}

File: backend/RecoveryREST/Models/Classes/RecoveryPlan.cs
namespace RecoveryREST.Models.Classes {
    public class RecoveryPlan {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string AppUserId { get; set; } = string.Empty;
        public AppUser AppUser { get; set; }
        public List<WorkoutDay> WorkoutDays { get; set; } = new List<WorkoutDay>();
        public bool IsCreatedByDoctor { get; set; } = false;
        public string? DoctorId { get; set; }
        public AppUser? Doctor { get; set; }
    }
}